docs/source/plotly_directive.py

```diff
@@ -0,0 +1,151 @@
+"""
+Usage:
+```
+.. plotly::
+
+    import plotly.graph_objects as go
+    import numpy as np
+
+    x = np.linspace(0, 10, 100)
+    y = np.sin(x)
+
+    go.Figure(data=go.Scatter(x=t, y=y, mode='markers'))
+```
+
+Requirements:
+1. docstring contains a single code block.
+2. the code block ends with an expression that evaluates to a plotly figure.
+
+Based on:
+https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/sphinxext/plot_directive.py
```

Comment:
Matplotlib's plot_directive is much more complicated because it supports multile code blocks and figures.